bindings/types.go

```diff
@@ -0,0 +1,194 @@
+package bindings
+
+import (
+	"net/url"
+	"sort"
+	"text/template"
+
+	"github.com/onflow/flixkit-go"
+	v1_1 "github.com/onflow/flixkit-go/flixkitv1_1"
+	"github.com/stoewer/go-strcase"
+)
+
+type simpleParameter struct {
+	Name        string
+	JsType      string
+	Description string
+	FclType     string
+	CadType     string
+}
+
+type templateData struct {
+	FclVersion           string
+	Version              string
+	Parameters           []simpleParameter
+	ParametersPrefixName string
+	Output               simpleParameter
+	Title                string
+	Description          string
+	Location             string
+	IsScript             bool
+	IsLocalTemplate      bool
+}
+
+type FclGenerator struct {
+	Templates []string
+}
+
+type FlixParameter struct {
+	Name string
+	Type string
+}
+
+func isUrl(str string) bool {
+	u, err := url.Parse(str)
+	return err == nil && u.Scheme != "" && u.Host != ""
+}
+
+func GetFlixFclCompatibility(flixVersion string) string {
+	compatibleVersions := map[string]string{
+		"1.0.0": "1.3.0",
```

Comment:
we probably should have the capabile versions (1.0 and 1.1) as constants somewhere to refer back to, which also should be referred inside the switch case comment i made above